publish_data.go

```diff
@@ -96,8 +117,8 @@ func (dp DataPublisher) PublishData(records []*DataRecord) error {
 	if dp.HasPubFeederChan() {
 		dp.PubFeederChan <- records
 	}
-	for _, record := range records {
-		if dp.HasLJH22() {
+	if dp.HasLJH22() {
```

Comment:
Yes, definitely do a loop over records inside each branch of the form `if dp.HasWhatever() {...}` The new way is smarter.